src/main/java/org/apache/flink/benchmark/SortingBoundedInputBenchmarks.java

```diff
@@ -0,0 +1,418 @@
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.datastream.KeyedStream;
+import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperatorV2;
+import org.apache.flink.streaming.api.operators.BoundedMultiInput;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.ChainingStrategy;
+import org.apache.flink.streaming.api.operators.Input;
+import org.apache.flink.streaming.api.operators.MultipleInputStreamOperator;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.api.operators.StreamOperator;
+import org.apache.flink.streaming.api.operators.StreamOperatorFactory;
+import org.apache.flink.streaming.api.operators.StreamOperatorParameters;
+import org.apache.flink.streaming.api.operators.TwoInputStreamOperator;
+import org.apache.flink.streaming.api.transformations.KeyedMultipleInputTransformation;
+import org.apache.flink.streaming.runtime.streamrecord.LatencyMarker;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.SplittableIterator;
+
+import org.junit.Assert;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+/**
+ * An end to end test for sorted inputs for a keyed operator with bounded inputs.
+ */
+public class SortingBoundedInputBenchmarks extends BenchmarkBase {
+
+	protected static final int RECORDS_PER_INVOCATION = 900_000;
+
+	public static void main(String[] args) throws RunnerException {
+		Options options = new OptionsBuilder()
+				.verbosity(VerboseMode.NORMAL)
+				.include(".*" + SortingBoundedInputBenchmarks.class.getCanonicalName() + ".*")
+				.build();
+
+		new Runner(options).run();
+	}
+
+	@State(Thread)
+	public static class SortingInputContext extends FlinkEnvironmentContext {
+		@Param({"true", "false"})
+		public boolean useSingleKeyStateBackend;
```

Comment:
Hmmm, do we need to compare it against non single key state backends (`useSingleKeyStateBackend = false`)? (We already have benchmarks for one/two/multiple input operators) ?

Reply:
The difference is in a state backend that is used. In both cases I do sort the inputs, so it is different from plain benchmarks for one/two/multiple operators. The difference between the "special" state backend vs memory state backend is not big enough to manifest in the tests. Therefore I removed the flag as per your suggestion.